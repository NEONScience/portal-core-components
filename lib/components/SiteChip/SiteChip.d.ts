export default WrappedSiteChip;
declare const WrappedSiteChip: any;
