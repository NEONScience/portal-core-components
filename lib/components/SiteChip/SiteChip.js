"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _styles = require("@material-ui/core/styles");
var _Chip = _interopRequireDefault(require("@material-ui/core/Chip"));
var _Avatar = _interopRequireDefault(require("@material-ui/core/Avatar"));
var _Place = _interopRequireDefault(require("@material-ui/icons/Place"));
var _Cancel = _interopRequireDefault(require("@material-ui/icons/Cancel"));
var _Theme = _interopRequireDefault(require("../Theme/Theme"));
const _excluded = ["label"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.includes(n)) continue; t[n] = r[n]; } return t; }
const useStyles = (0, _styles.makeStyles)(theme => ({
  avatarLarge: {
    '& svg': {
      height: theme.spacing(3)
    }
  }
}));
const useChipStyles = (0, _styles.makeStyles)(theme => ({
  outlined: {
    color: theme.palette.grey.A200
  },
  outlinedPrimary: {
    color: theme.palette.primary.main
  },
  outlinedSecondary: {
    color: theme.palette.secondary.main
  }
}));
const SiteChip = props => {
  const classes = useStyles(_Theme.default);
  const chipClasses = useChipStyles(_Theme.default);
  const {
      label
    } = props,
    otherProps = _objectWithoutPropertiesLoose(props, _excluded);

  // Default optional props
  if (!otherProps.color) {
    otherProps.color = 'secondary';
  }
  if (!otherProps.variant) {
    otherProps.variant = 'outlined';
  }
  if (!otherProps.size) {
    otherProps.size = 'small';
  }
  if (!otherProps['data-selenium']) {
    otherProps['data-selenium'] = 'site-chip';
  }

  /*
    Avatar style overrides
    Material UI has a bug where Avatars in chips do not inherit styles correctly
    in production due to CSS class definition order and strict specificity. See here:
    https://github.com/mui-org/material-ui/issues/16374
    Should that get fixed upstream then all inline styles on <Avatar> can be removed.
    There is also no 'large' size despite having a small and medium, and we have
    use cases for large (see DataProductAvailability).
  */
  const avatarStyle = {
    width: '32px',
    height: '32px',
    marginLeft: '0px'
  };
  if (['primary', 'secondary'].includes(otherProps.color)) {
    avatarStyle.backgroundColor = _Theme.default.palette[otherProps.color].main;
  }
  if (otherProps.size === 'small') {
    avatarStyle.width = '24px';
    avatarStyle.height = '24px';
  }

  /*
    General style overrides
    There is also 'large' size despite having a small and medium, and we have
    use cases for large (see DataProductAvailability).
  */
  const chipStyle = {};
  const deleteIconStyle = {};
  let avatarClass = null;
  if (otherProps.size === 'large') {
    otherProps.size = null; // to prevent a PropTypes error in Mui/Chip
    avatarClass = classes.avatarLarge;
    chipStyle.fontSize = '1rem';
    chipStyle.borderRadius = '20px';
    chipStyle.height = '40px';
    avatarStyle.width = '40px';
    avatarStyle.height = '40px';
    deleteIconStyle.width = '32px';
    deleteIconStyle.height = '32px';
    deleteIconStyle.margin = '0 6px 0 0';
  }
  return /*#__PURE__*/_react.default.createElement(_Chip.default, _extends({}, otherProps, {
    label: label,
    style: chipStyle,
    avatar: /*#__PURE__*/_react.default.createElement(_Avatar.default, {
      className: avatarClass,
      style: avatarStyle
    }, /*#__PURE__*/_react.default.createElement(_Place.default, null)),
    deleteIcon: /*#__PURE__*/_react.default.createElement(_Cancel.default, {
      style: deleteIconStyle
    }),
    classes: chipClasses
  }));
};
SiteChip.propTypes = {
  label: _propTypes.default.string.isRequired
};
const WrappedSiteChip = _Theme.default.getWrappedComponent(SiteChip);
var _default = exports.default = WrappedSiteChip;