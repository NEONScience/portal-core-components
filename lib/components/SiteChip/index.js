"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function () {
    return _SiteChip.default;
  }
});
var _SiteChip = _interopRequireDefault(require("./SiteChip"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }