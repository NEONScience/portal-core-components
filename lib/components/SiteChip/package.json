{
  "private": true,
  "name": "site-chip",
  "main": "./SiteChip.jsx",
  "module": "./SiteChip.js"
}
