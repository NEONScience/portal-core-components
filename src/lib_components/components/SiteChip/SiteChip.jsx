import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@mui/styles';
import Chip from '@mui/material/Chip';
import Avatar from '@mui/material/Avatar';
import PlaceIcon from '@mui/icons-material/Place';
import DeleteIcon from '@mui/icons-material/Cancel';

import Theme from '../Theme/Theme';

const useStyles = makeStyles((theme) => ({
  avatarLarge: {
    '& svg': {
      height: theme.spacing(3),
    },
  },
}));

const useChipStyles = makeStyles((theme) => ({
  outlined: {
    color: theme.palette.grey.A200,
  },
  outlinedPrimary: {
    color: theme.palette.primary.main,
  },
  outlinedSecondary: {
    color: theme.palette.secondary.main,
  },
}));

const SiteChip = (props) => {
  const classes = useStyles(Theme);
  const chipClasses = useChipStyles(Theme);
  const { label, ...otherProps } = props;

  // Default optional props
  if (!otherProps.color) { otherProps.color = 'secondary'; }
  if (!otherProps.variant) { otherProps.variant = 'outlined'; }
  if (!otherProps.size) { otherProps.size = 'small'; }
  if (!otherProps['data-selenium']) { otherProps['data-selenium'] = 'site-chip'; }

  // Backwards compatible with MUI v4
  if (otherProps.variant === 'default') {
    otherProps.variant = 'filled';
  }

  /*
    Avatar style overrides
    Material UI has a bug where Avatars in chips do not inherit styles correctly
    in production due to CSS class definition order and strict specificity. See here:
    https://github.com/mui-org/material-ui/issues/16374
    Should that get fixed upstream then all inline styles on <Avatar> can be removed.
    There is also no 'large' size despite having a small and medium, and we have
    use cases for large (see DataProductAvailability).
  */
  const avatarStyle = {
    width: '32px',
    height: '32px',
    marginLeft: '0px',
  };
  if (['primary', 'secondary'].includes(otherProps.color)) {
    avatarStyle.backgroundColor = Theme.palette[otherProps.color].main;
  }
  if (otherProps.size === 'small') {
    avatarStyle.width = '24px';
    avatarStyle.height = '24px';
  }

  /*
    General style overrides
    There is also 'large' size despite having a small and medium, and we have
    use cases for large (see DataProductAvailability).
  */
  const chipStyle = {};
  if (otherProps.variant === 'outlined') {
    chipStyle.backgroundColor = 'transparent';
  } else if ((otherProps.variant === 'filled') && (otherProps.color === 'secondary')) {
    chipStyle.backgroundColor = Theme.palette.secondary.main;
  } else if ((otherProps.variant === 'filled') && (otherProps.color === 'primary')) {
    chipStyle.backgroundColor = Theme.palette.primary.main;
  }
  const deleteIconStyle = {};
  let avatarClass = null;
  if (otherProps.size === 'large') {
    avatarClass = classes.avatarLarge;
    chipStyle.fontSize = '1rem';
    chipStyle.borderRadius = '20px';
    chipStyle.height = '40px';
    avatarStyle.width = '40px';
    avatarStyle.height = '40px';
    deleteIconStyle.width = '32px';
    deleteIconStyle.height = '32px';
    deleteIconStyle.margin = '0 6px 0 0';
  }

  return (
    <Chip
      {...otherProps}
      label={label}
      style={chipStyle}
      avatar={<Avatar className={avatarClass} style={avatarStyle}><PlaceIcon /></Avatar>}
      deleteIcon={<DeleteIcon style={deleteIconStyle} />}
      classes={chipClasses}
    />
  );
};

SiteChip.propTypes = {
  label: PropTypes.string.isRequired,
};

const WrappedSiteChip = Theme.getWrappedComponent(SiteChip);

export default WrappedSiteChip;
