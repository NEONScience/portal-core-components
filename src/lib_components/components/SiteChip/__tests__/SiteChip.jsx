import React from 'react';
import renderer from 'react-test-renderer';

import SiteChip from '../SiteChip';

describe('SiteChip', () => {
  test('Renders correctly', () => {
    const tree = renderer
      .create(<SiteChip label="foo" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
