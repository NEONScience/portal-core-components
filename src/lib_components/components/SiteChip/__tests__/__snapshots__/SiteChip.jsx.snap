// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteChip Renders correctly 1`] = `
<div
  className="MuiChip-root MuiChip-colorSecondary MuiChip-outlined makeStyles-outlined-2 MuiChip-outlinedSecondary makeStyles-outlinedSecondary-4 MuiChip-sizeSmall"
  data-selenium="site-chip"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  style={Object {}}
>
  <div
    className="MuiAvatar-root MuiAvatar-circle MuiChip-avatar MuiChip-avatarSmall MuiChip-avatarColorSecondary MuiAvatar-colorDefault"
    style={
      Object {
        "backgroundColor": "#0073cf",
        "height": "24px",
        "marginLeft": "0px",
        "width": "24px",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
      />
    </svg>
  </div>
  <span
    className="MuiChip-label MuiChip-labelSmall"
  >
    foo
  </span>
</div>
`;
